src/pages/iou/withMoneyRequest.js

```diff
@@ -0,0 +1,93 @@
+import React, {createContext, forwardRef, useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import getComponentDisplayName from '../../libs/getComponentDisplayName';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../../components/withCurrentUserPersonalDetails';
+import * as PersonalDetails from '../../libs/actions/PersonalDetails';
+import optionPropTypes from '../../components/optionPropTypes';
+import useOnNetworkReconnect from '../../components/hooks/useOnNetworkReconnect';
+
+const MoneyRequestContext = createContext();
+const moneyRequestPropTypes = PropTypes.shape({
+    // Money request amount
+    amount: PropTypes.number.isRequired,
+
+    // Money request currency
+    currency: PropTypes.string.isRequired,
+
+    // Selected participants from MoneyRequestParticipantsPage with login
+    participants: PropTypes.arrayOf(optionPropTypes),
+
+    // Money request description
+    comment: PropTypes.string.isRequired,
+
+    // Callback to update money request amount
+    setAmount: PropTypes.func.isRequired,
+
+    // Callback to update money request currency
+    setCurrency: PropTypes.func.isRequired,
+
+    // Callback to update money request participants
+    setParticipants: PropTypes.func.isRequired,
+
+    // Callback to update money request description
+    setComment: PropTypes.func.isRequired,
+});
+
+const MoneyRequestProvider = (props) => {
+    const [amount, setAmount] = useState(0);
+    const [currency, setCurrency] = useState(props.currentUserPersonalDetails.localCurrencyCode);
+    const [participants, setParticipants] = useState([]);
+    const [comment, setComment] = useState('');
+
+    useEffect(() => {
+        PersonalDetails.openMoneyRequestModalPage();
+    }, []);
```

Comment:
This is missing a call to setCurrency.

Reply:
No need to call setCurrency as we already set it in the currency initial value.
https://github.com/Expensify/App/pull/17964/files/e7521264711339bbdb92363b82ff09a148f3655b#diff-8e9e82317b35e5d738c66bfb0541371c679db296e1793e7c1601a02aa8a546a0R38

Reply:
Wasn't the call made to ensure that we have `currentUserPersonalDetails.localCurrencyCode`. Otherwise what's the point of calling `PersonalDetails.openMoneyRequestModalPage()`?

Reply:
The call will be async, right? So, I think calling setCurrency will just set the same value we already have. `currentUserPersonalDetails.localCurrencyCode` is ready in the Onyx, but I should have added a fallback in case the user access the page while all the data is not loaded yet (after login, you can reproduce this one in staging and it will throw currency undefined error). The reason maybe to just update the currency code in the onyx, so the next time we open the page, it will show the new currency.

Reply:
Right, I noticed that. I don't know the use of `PersonalDetails.openMoneyRequestModalPage` (both on mount and on reconnect).

@thienlnam You added that logic here https://github.com/Expensify/App/commit/edd210c03fb81e7a97f43ce79f069cac2c91979c. Can you please explain the use of that function call. After making the call we don't make use of the new `currentUserPersonalDetails.localCurrencyCode` value.

Reply:
I just refactored the logic that was already there to use hooks but I'm guessing like @s77rt mentioned it was to ensure that the value existed because of this problem existing
> I should have added a fallback in case the user access the page while all the data is not loaded yet (after login, you can reproduce this one in staging and it will throw currency undefined error)

Reply:
Looks like the call is useless. We don't re-render after making the call and getting new values. I think we should just remove it.

Reply:
Just to confirm, are we safe to completely remove `PersonalDetails.openMoneyRequestModalPage`? I'm just afraid there are other things going on in the API besides returning the user local currency code 😄.

Reply:
I understand your concern here. I am afraid I don't have a clear answer to this.

- Frontend: I doubt we are breaking anything here, the call response (onyx data) was not taken into consideration (unless you reopen the modal - I don't see any case where that would be useful).
- Backend: @thienlnam or @PauloGasparSv Can you please confirm that OpenIOUModalPage command do nothing but return the currency code?


If we feel unsure about that we can leave the API calls. I'm just inclined towards removal to clean the logic (remove redundant calls).

Reply:
I appreciate the concern with not breaking things! I took a look and all OpenIOUModalPage does in the BE is push onyx  updates with the local currency

Reply:
@thienlnam Thanks for the confirmation.

> unless you reopen the modal

@s77rt this is actually the thing that I think it's useful in case the currency of the user could change while they are using the app. If we only rely from the OpenApp, user need to reopen the app to get their updated currency. I think ReconnectApp will also give us the currency, so we can put the app to the background and come back to trigger it.

Actually it's not really a big thing, so do you think we are okay with that?

Reply:
It's hard to imagine a scenario where this call is actually doing anything helpful :sweat_smile:. Let's remove it and test various cases.

Reply:
Let's mark this as resolved and focus on the other issue.